pages/rfc/registry-updates.rst

```diff
@@ -0,0 +1,94 @@
+.. _nuts-documentation-registry-updates:
+
+RFC: Registry Updates
+#####################
+
+Motivation
+**********
+
+The Nuts Registry contains information about who (vendors, organizations) participates in the Nuts network and where
+to find the resources they're exposing through endpoint registration. Parties also use it to register their certificates
+so that other parties can verify the integrity and authenticity of exposed data.
+
+The registry is modelled as an append-only event log which makes it easy to add new data (e.g. register an endpoint).
+However, there is currently no way to alter an data (e.g. rename an organization). This is an essential feature for any
+Nuts network which is managed by more than a single party, since altering data currently requires coordinated data
+pruning across all participating Nuts nodes.
+
+This document proposes a mechanism for updating data in the registry.
+
+Status of this RFC
+******************
+
+DRAFT
+
+Terminology
+***********
+
+- Event: an event in the registry event log, e.g. the RegisterVendor event
+- Object: the thing an event describes; either a vendor, organization or endpoint.
+- Event header: top-level properties of an event describing its payload.
+
+Mechanics
+*********
+
+Since events itself are immutable, only new events can be added to describe an object's new state. A subsequent event
+completely replaces the object's state described by the preceding event. The subsequent event must refer to the event
+it revises by its hash.
+
+Event referral
+==============
+
+An event refers to a preceding event (which it replaces) by its hash. There can only be one event referring to a particular
+event. In other words, subsequent events form a single stream without branching/merging:
+
+.. code-block:: console
+
+    +----+ replaces +----+ replaces +----+
+    |e(1)+<---------+e(2)+<---------+e(3)|
+    +--+-+          +----+          +--+-+
+       ^                               ^
+       |                               |
+    initial                         current
+
+An event's hash is calculated by hashing its payload. The hash itself is specified the event's header in the ``hash`` field.
+If there's a preceding event its hash is specified in the ``prev`` field:
+
+.. code-block:: json
+
+    {
+      "hash": "...",
+      "prev": "...",
+      (etc)
+      "payload": {
+        "identifier": "a09039da-f96b-4fd1-8925-fd28cb833159",
+        "prop1": "Hello",
+        "prop2": "World"
+      }
+    }
+
+Events should specify the ``hash`` header as an extra consistency check. When processing an event, its hash should be
+calculated and compared to the ``hash`` header. If it doesn't match the event must be rejected.
+For historic events which do not contain the ``hash`` header this check is skipped.
+
+Hashing
+^^^^^^^
+
+An event is hashed by taking its payload and calculating a SHA-1 hash over its stringified payload. To maintain a
```

Comment:
why not sha-256

Reply:
I'd think SHA-1 is good enough to avoid accidental collisions between event payloads; it's also not used for cyptographic integrity checking since that's what the JWS is for.

While we can always use longer hashes they come at the expense of larger messages/slower computation.